packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -251,13 +267,22 @@ contract Bridge is EssentialContract, IBridge {
             receipt = ProofReceipt(uint64(block.timestamp), 0);
 
             if (invocationDelay != 0) {
-                // if (!transactedByOwner) {
-                //     receipt.feePaid = 0;
-                //     msg.sender.sendEtherAndVerify(receipt.feePaid, _SEND_ETHER_GAS_LIMIT);
-                // }
+                if (!transactedByOwner) {
+                    receipt.feePaid = uint160(
+                        _calcFee(
+                            _message.fee,
+                            _message.gasLimit,
+                            GAS_RECEIVING,
```

Comment:
I think having a fixed gas cost here can be very inaccurate because merkle proofs can have very different costs. I don't know in practice how much they actually differ in costs at roughly the same tree size (very much depends which data and how the tree looks like), but the longer the signal contract is used the larger the proofs will get on average

The other difficulty is that a large part of the gas cost will be related to the calldata cost of the proof, not the processing cost inside the smart contract. Not sure if `GAS_PROCESSING` takes that into account, but seems like a pretty low number (though because of the small Merkle proofs when not a lot of data is already stored in the smart contract, maybe accurate because of that reason). 

For an accurate cost for the receiving, I would measure the gas used for verifying the merkle proof, and additional charge `proof.length * 16` gas for the proof calldata.